### PR TITLE
DDO-3517: add eso secrets to data repo

### PR DESCRIPTION
**Pull request by katiewelch**
This is the first of 4 PRs to migrate data repo to ESO from Vault. This PR and [DDO-3517 in terra-helmfile](https://github.com/broadinstitute/terra-helmfile/pull/5213) will add eso secrets to data repo. The ESO secrets will not be used yet, they will sit along side the Vault secrets until PRs 3 & 4.

This repo (datarepo-helm) will hold the create-secret-manager-secret chart while terra-helmfile will hold the values.

This PR was tested by using [the datarepo-helm-definitions repo](https://github.com/broadinstitute/datarepo-helm-definitions). Using Olivia's setup, I pointed the chart to my local updated version of the create-secret-manager-secrets chart, after updating the values file in the definitions repos. I also pointed the values to my local versions of the terra-helmfile values files for dev, staging, prod, and BEEs.

helmfile.yaml:
```
# helm releases to be deployed
releases:
  - name: ok-jade-create-secret-manager-secret    # release name
    namespace: ok   # target namespace
    createNamespace: true
    #chart: datarepo-helm/create-secret-manager-secret   # chart name    
    chart: /Users/welchkat/Documents/src/datarepo-helm/charts/create-secret-manager-secret   # chart name
    missingFileHandler: Warn
    values:
      #- create-secret-manager-secret.yaml      # Value files passed via --values
      - /Users/welchkat/Documents/src/terra-helmfile/values/app/datarepo/live/dev.yaml
```

After running `helmfile apply`, for each iteration of my local chart and values files, I confirmed the output matches what I expect.

### CODE REVIEW

File: charts/create-secret-manager-secret/values.yaml [hunk 1]
```diff
@@ -9,3 +9,15 @@ secrets:
...
+  secrets: []
+  refreshInterval: "5m"
+
+# - secretName:               ## name of kubeSecret
+#   data:
+#     - secretKey:            ## name of kubeSecretKey
+#       remoteRef:
+#         key:                ## key name
+#         property:           ## description of secret
+#         decodingStrategy:   ## how to decode secret
```
Comment (jack-r-warren) [inline charts/create-secret-manager-secret/values.yaml:23]: I'm not sure this is accurate -- I think it's
```
#     - secretKey:            ## name of kubeSecretKey
#       key:                ## key name
#       property:           ## description of secret
#       decodingStrategy:   ## how to decode secret
```

File: charts/create-secret-manager-secret/templates/vaultsecret.yaml [hunk 2]
```diff
@@ -19,3 +19,27 @@ spec:
...
     {{- end }}
 {{- end -}}
+
+{{- range .Values.externalSecrets.secrets }}
+---
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: {{ .secretName }}
+  labels: {{- include "datarepo.labels" . | nindent 4 }}
```
Comment (jack-r-warren) [inline charts/create-secret-manager-secret/templates/vaultsecret.yaml:29]: I don't think there are labels here -- there might be? Not sure, the old secretDefintion didn't have them

File: charts/create-secret-manager-secret/templates/vaultsecret.yaml
Comment (snf2ye) [inline charts/create-secret-manager-secret/templates/vaultsecret.yaml:1]: You'll need to manual bump the version on any of the charts changed:
https://github.com/broadinstitute/datarepo-helm/blob/master/charts/create-secret-manager-secret/Chart.yaml#L17

Comment (katiewelch) [reply charts/create-secret-manager-secret/templates/vaultsecret.yaml:1]: Done

File: charts/create-secret-manager-secret/Chart.yaml [hunk 3]
```diff
@@ -14,7 +14,7 @@ type: application
 
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 0.0.6
+version: 0.0.7
 appVersion: 0.0.6
```
Comment (snf2ye) [inline charts/create-secret-manager-secret/Chart.yaml:18]: I think we also bump the appVersion to match... I don't know why we have both.

**Review by snf2ye — APPROVED** [review]

**Pull request merged**